### ISSUE TITLE
The Multiaddr should not be used as the key for the ConnectionReuse

### ISSUE THREAD

**Issue by tomaka** [libp2p/rust-libp2p#307]
For Unix domain sockets, we may connect to many different peers whose multiaddress will all be `/unix/<path>`. Therefore there's no way to distinguish them with this.

**Comment by tomaka** [comment]
No longer relevant after #565
